packages/swapper/src/swappers/utils/helpers/helpers.ts

```diff
@@ -168,3 +176,18 @@ export const normalizeIntegerAmount = (amount: string | number | BN): string =>
     .toNumber()
     .toLocaleString('fullwide', { useGrouping: false })
 }
+
+export const getApproveContractData = async ({
+  web3,
+  sellAssetErc20Address,
+  receiveAddress
+}: GetApproveContractDataArgs): Promise<string> => {
+  const approveContract = new web3.eth.Contract(erc20AbiImported, sellAssetErc20Address)
+
+  const preApprove = await approveContract.methods.approve(
+    COW_SWAP_VAULT_RELAYER_ADDRESS,
```

Comment:
The spender address should be passed in seeing as this is a generic helper

Reply:
Fixed ! thanks